### ISSUE TITLE
Summarizing multiple training histories

### ISSUE THREAD

**Issue by gsabot** [rstudio/keras#479]
Hi, I'm working on a problem where I have a large collection of
related training history data, and I'm wondering if there are any
keras/R tools to help summarize and/or view that combined history.
I'm not looking to simply generate a grid of n related but independent
graphs from plot.keras_training_history().  Rather, I'm looking for
something more general to help me understand what the "typical"
training history was across a set of related models.

The context is a backtest that predicts a time series.  For a
straightforward backtest example, see this article:

  https://blogs.rstudio.com/tensorflow/posts/2018-06-25-sunspots-lstm/

As in the sunspot example, I have a series of models generated on
sliding windows that expose more and more of the underlying data over
time.  This kind of thing crops up in many time series contexts.  The
common element: each rollforward window generates a "freshly trained"
model, with its own training history, but all the models are related
in having the same architecture and in addressing the same problem.
This same issue can also happen cross sectionally, when you train
separate models for various problems that share some common element.

The key is that the a "good" approach to training the architecture
should measure out as good on "most" of the models, but likely not on
all of them.  You would not want to tune your training hyperparameters
while looking at just one model, you instead want a feel for typical
training history, outliers, etc.

My first thought was to take the various metric histories and compute
summary info, such as the mean loss across all models at epoch 1, 2,
3, etc, percentiles, etc, and then display a graph or table of those
combined metrics.  But there are many issues that glosses over (is
some "history normalization" appropriate, how to handle NAs due to
early stopping in some models, etc)

Can you offer any pointers to tools or research touching on these
multi-model issues, or perhaps advice on alternative ways to think
about them?

Thanks.

**Comment by jjallaire** [comment]
The tfruns package overlaps to some extent with what you are describing here: https://tensorflow.rstudio.com/tools/tfruns/articles/overview.html

**Comment by skeydan** [comment]
And for the approach... it could be interesting to compare distributions of metrics, dependent on the hyperparameter settings ... e.g., say that with dropout=0.3 you get higher variance in relevant metrics (the sample being all windows used to train the model on with this particular setting) than with say dropout=0.4 (just an example, but could be interesting)

**Comment by gsabot** [comment]
Thanks, tfruns sounds like it could be useful starting point for me.
